evm/db/hash_trie.py

```diff
@@ -1,28 +1,10 @@
-from eth_hash.auto import keccak
-
-
-class HashTrie(object):
-    _trie = None
-
-    def __init__(self, trie):
-        self._trie = trie
+from functools import partial
 
-    def __setitem__(self, key, value):
-        self._trie[keccak(key)] = value
-
-    def __getitem__(self, key):
-        return self._trie[keccak(key)]
-
-    def __delitem__(self, key):
-        del self._trie[keccak(key)]
+from eth_hash.auto import keccak
 
-    def __contains__(self, key):
-        return keccak(key) in self._trie
+from evm.db.keymap import (
+    KeyMapDB,
+)
 
-    @property
-    def root_hash(self):
-        return self._trie.root_hash
 
-    @root_hash.setter
-    def root_hash(self, value):
-        self._trie.root_hash = value
+HashTrie = partial(KeyMapDB, keccak)
```

Comment:
This is probably going to be problematic when it comes to applying type hints.  Should probably figure out a way to do this with proper subclassing.

Reply:
I guess this should do it:

```py
class HashTrie(KeyMapDB):
    def __init__(self, db):
        super().__init__(keccak, db)
```

Reply:
I'd do it as a class attribute.

```python
class KeyMapDB:
    @staticmethod
    @abstractmethod
    def key_fn(value: bytes) -> bytes:
        raise NotImplementedError

class HashTrie(KeyMapDB):
    key_fn = staticmethod(keccak)
```

Reply:
Huh, mypy complained about wrapping `keccak` in `staticmethod()`. Works fine without it.